### PR TITLE
Add `istio.io/meshID` annotation to injection templates

### PR DESCRIPTION
**Pull request by jewertow**
**Please provide a description of this PR:**

This PR adds injection of `istio.io/meshID` to all istio-proxies: sidecars, default gateways, injected gateways, k8s api gateways, grpc-agent and waypoint.

This annotation will distinct metrics collected in multi-tenant meshes or in multi-cluster meshes with global Prometheus.

### CODE REVIEW

**Comment by kebe7jun** [comment]
If you just want to get the meshid from the metrics, maybe this can be achieved by configuring telemetry?
Like:

```
  telemetry:
    v2:
      prometheus:
        configOverride:
          inboundSidecar:
            metrics:
              - dimensions:
                  destination_mesh_id: "node.metadata['MESH_ID']"
                  source_mesh_id: "downstream_peer.mesh_id"
                  destination_cluster: "node.metadata['CLUSTER_ID']"
                  source_cluster: "downstream_peer.cluster_id"

```
This is how we currently use it.

**Comment by jewertow** [comment]
> I not sure what's a global Prometheus? can you share more details?

@zirain I saw configurations where users have multiple clusters with short retention Prometheus instances and one cluster with long retention to which all others write scraped metrics. In such configurations mesh id would make it possible to distinct source of metrics.

> IMO, it's easy to achieve when Prometheus relabel? It's not necessary to add a new annotation.

Yes, this is an alternative, but I wanted to make it more seamless and transparent to users.

**Comment by jewertow** [comment]
@kebe7jun thanks! I will try this approach. If I won't find any limitation of this solution, I will close this PR.

**Comment by kyessenov** [comment]
@jewertow Make sure to use Telemetry API on the recent versions instead of operator config.

**Review by ericvn — CHANGES_REQUESTED** [review]
I don't see istio.io/meshID in the annotations defined in istio/api. Without it defined there, the istioctl analyzer will generate an unknown annotation message and this will fail at least one istio.io test. Hold this PR out until that annotation is merged.

**Comment by costinm** [comment]
Sorry I missed this - since it's really an API change we need to discuss it
first, not sure if injection is the right place ( Gateways created by
istiod, VMs, etc), don't think meshID is the
right way to identify tenants, etc. Also we need to consider the broader
context - is this something that can be done by the telemetry provider or
independent of Istio ?

**Comment by costinm** [comment]
Perfect - that was what I meant, relabeling, otel collector, etc - there
are many good ways to enhance the metrics with more information without
having the complexity in istio.

**Pull request closed without merging**
